SofaKernel/modules/SofaBaseLinearSolver/src/SofaBaseLinearSolver/FullVector.inl

```diff
@@ -36,7 +36,7 @@ void FullVector<Real>::checkIndex(Index n) const
     {
         msg_error("FullVector") << "in vector<" << sofa::helper::gettypename(typeid(*this)) << "> " << std::hex << (long)this << std::dec << " size " << cursize << " : invalid index " << (int)n;
         sofa::helper::BackTrace::dump();
-        assert(i < size);
+        assert(n < cursize);
```

Comment:
how can this assert happen if inside this if: https://github.com/sofa-framework/sofa/blob/47cde44a64dbc81c54e7f10d0deb17edcebaa1f2/SofaKernel/modules/SofaBaseLinearSolver/src/SofaBaseLinearSolver/FullVector.inl#L35

Reply:
http://www.cplusplus.com/reference/cassert/assert/

```
If the argument expression of this macro with functional form compares equal to zero (i.e., the expression is false), a message is written to the standard error device and abort is called, terminating the program execution.
```
it is made that it will always assert;
and that was the code present before actually 😗

Reply:
ah yes... sorry I always read assert in the bad sens